### PR TITLE
PLANNER-1021 Incorporate suggestions + cleanup

### PR DESCRIPTION
**Pull request by yurloc**

### CODE REVIEW

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by yurloc** [comment]
@rsynek There was a couple of suggestions in the original PR for PLANNER-1021 (#1375) that didn't prevent it from being merged. We wanted to implement those suggestions without burdening the contributor. One of the suggestions is addressed by the commit with PLANNER-1021 in its subject. The other suggestion was formatting related. I already had some formatting and cleanup work in progress so I used this occasion to finalize it.

Would you prefer to have the cleanup work in a separate PR?

**Comment by rsynek** [comment]
@yurloc no, that's completely fine - I just missed the entire picture, thanks for pointing me to the original PR. Thanks

**Comment by yurloc** [comment]
@rsynek OK. I should have written at least the basic description :slightly_smiling_face:

**Comment by yurloc** [comment]
@ge0ffrey Please merge.

File: kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/SolverServiceBase.java [hunk 1]
```diff
@@ -480,31 +480,28 @@ private void solvePlanningProblem(final SolverInstanceContext sic,
...
+                    // TODO Race condition: status turns into non-solving before solver starts
+                    // See https://issues.jboss.org/browse/PLANNER-540
+                }
+                if (status == SolverInstance.SolverStatus.SOLVING) {
+                    sic.getSolver().solve(planningSolution);
+                }
+            } catch (Exception e) {
+                logger.error("Exception executing solver '" + sic.getInstance().getSolverId()
+                                     + "' from container '" + sic.getInstance().getContainerId() + "'. Thread will terminate.",
+                             e);
```
Comment (ge0ffrey) [inline kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/SolverServiceBase.java]: Can you open a jira that we need to report that error back to the user when he comes asking for the result in a new REST request?

**Pull request merged**